app/src/main/java/mozilla/lockbox/LockboxAutofillService.kt

```diff
@@ -104,6 +107,7 @@ class LockboxAutofillService(
                     is AutofillAction.Complete -> builder.buildFilteredFillResponse(this, listOf(it.login)).asOptional()
                     is AutofillAction.CompleteMultiple -> (builder.buildFilteredFillResponse(this, it.logins)
                         ?: builder.buildFallbackFillResponse(this)).asOptional()
+                    is AutofillAction.Filter -> builder.buildFallbackFillResponse(this).asOptional()
```

Comment:
Does `asOptional` make sense here? 

Since we're putting `asOptional` on the outside of the `when` clause?

Reply:
i.e.

```kotlin
when (it) {
    is AutofillAction.Complete -> builder.buildFilteredFillResponse(this, listOf(it.login))
    is AutofillAction.CompleteMultiple -> (builder.buildFilteredFillResponse(this, it.logins)
        ?: builder.buildFallbackFillResponse(this))
    is AutofillAction.Filter -> builder.buildFallbackFillResponse(this)
    is AutofillAction.Authenticate -> builder.buildAuthenticationFillResponse(this)
    is AutofillAction.Cancel -> null
    is AutofillAction.Error -> {
        callback.onFailure(getString(R.string.autofill_error_toast, it.error.localizedMessage))
        null
    }
}.asOptional()
```

Reply:
I'm also a little bit unsettled by:

1. a `Filter` action being fulfilled by a fallback response.
2. a `Filter` action essentially _routing_ the user to a different activity. As a name, "filter" is quite overloaded with meanings, and lacks the clarity that "authenticate" has. Suggest: `LaunchFilter`, `SearchFallback`… but more thought around naming is needed.